### PR TITLE
TT-11522 DX-1257  MDCB helm chart installation

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

[DX-1257](https://tyktech.atlassian.net/browse/DX-1257)

---

### Preview Link

<br>

[Tyk Control Plane Chart](https://deploy-preview-4405--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-control-plane-chart/)

[Tyk Data Plane Chart](https://deploy-preview-4405--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-data-plane-chart/)

### Description

1. tyk-control-plane moving out from Beta to Stable
2. update tyk-data-plane with instruction how to get connection string for on-prem setup
<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

[DX-1257]: https://tyktech.atlassian.net/browse/DX-1257?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

___

## **Type**
enhancement, documentation

___

## **Description**
- Updated Tyk Control Plane Chart documentation to reflect its move from Beta to Stable.
- Removed Beta notices and `--devel` flags from helm commands across Tyk Control Plane and Tyk Data Plane chart documentation.
- Added new sections and updated existing ones for obtaining and configuring Tyk Developer Portal and MDCB licenses.
- Enhanced MongoDB configuration details across all charts, including SSL support and the ability to set connection details via Kubernetes secrets.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-control-plane-chart.md</strong><dd><code>Update Tyk Control Plane Chart Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md
<li>Updated the title to remove the Beta tag.<br> <li> Removed the Beta notice and the <code>--devel</code> flag from helm commands.<br> <li> Added a new section for obtaining Tyk Developer Portal License.<br> <li> Updated MongoDB configuration details and added SSL support.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4405/files#diff-1bb3ae7ce6b5aec344682558fe28a0e91ef5504473fbed884ae4651089784cf7">+47/-38</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-data-plane-chart.md</strong><dd><code>Enhance Tyk Data Plane Chart Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md
<li>Added instructions for obtaining remote control plane connection <br>details from Tyk Control Plane chart.<br> <li> Updated the prerequisites and installation instructions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4405/files#diff-da211afa42cedf90ce80f9d59419356f09f0a76d37f56788b523bdf02b198e4f">+53/-3</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-stack-chart.md</strong><dd><code>Update Tyk Stack Chart Documentation for MongoDB and Admin </code><br><code>Configurations</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md
<li>Updated MongoDB configuration to include SSL support and connection <br>details via secrets.<br> <li> Enhanced admin user configuration details to include Developer Portal.<br> <br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4405/files#diff-f6e45bbf530fa763d587dec798584c8eb6294229ef353d3212ab01bf6b0a1d7c">+35/-26</a>&nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by dcs3spp** [comment]
@caroltyk Is the PR ready for review and would it be possible to arrange a peer review?

**Comment by carolwkc** [comment]
@komalsukhani could you help to peer review?

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 1]
```diff
@@ -633,6 +642,40 @@ This port lets MDCB allow standard health checks.
 It also defines the path for liveness and readiness probes.
 It is used to set `TYK_MDCB_HEALTHCHECKPORT`
 
+#### Enabling MDCB TLS
+
+Assuming that TLS certificates for the Tyk MDCB are available in the Kubernetes Secret `mdcb-tls-secret`, follow these steps to enable TLS:
+1. Set `tyk-mdcb.mdcb.tls.useSSL` to true.
+2. Set `tyk-mdcb.mdcb.tls.secretName to the name of the Kubernetes secret containing TLS certificates for the Tyk MDCB, in this case, `mdcb-tls-secret`.
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
2. Set `tyk-mdcb.mdcb.tls.secretName` to the name of the Kubernetes secret containing TLS certificates for the Tyk MDCB, in this case, `mdcb-tls-secret`.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 2]
```diff
@@ -633,6 +642,40 @@ This port lets MDCB allow standard health checks.
...
+
+```yaml
+tyk-mdcb:
+  mdcb:
+    tls:
+      # enables ssl for mdcb
+      useSSL: false
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
      useSSL: true
```

Comment (buraksekili) [reply tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Shouldn't it be true?

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 3]
```diff
@@ -633,6 +642,40 @@ This port lets MDCB allow standard health checks.
...
+      # location to pem encoded private key
+      certificateKeyFile: "/etc/certs/tls.key"
+
+      # location to pem encoded certificate
+      certificateCertFile: "/etc/certs/tls.crt"
+
+      # Sets the AES256 secret which is used to encode certificate private keys when
+      # they uploaded via certificate storage.
+      # It is used to set TYK_MDCB_SECURITY_PRIVATECERTIFICATEENCODINGSECRET
+      privateCertificateEncodingSecret: ""
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: This field is located under `global.mtls` section according to https://github.com/TykTechnologies/tyk-charts/blob/7691c47451c54d9c8afb2e1b1d9f59fd92739793/components/tyk-mdcb/values.yaml#L134-L138C5

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 4]
```diff
@@ -66,16 +60,17 @@ At a minimum, modify `values.yaml` for the following settings:
 2. [Set Mongo or PostgreSQL connection details](#set-mongo-or-postgressql-connection-details-required)
 3. [Tyk Dashboard License](#tyk-dashboard-license-required)
 4. [Tyk MDCB License](#tyk-mdcb-license-required)
+5. If you would like to use Developer Portal, an additional license is required: [Tyk Developer Portal License](#tyk-developer-portal-license-required)
 
-If you would like to use Developer Portal, an additional license is required:
-
-5. [Tyk Developer Portal License](#tyk-developer-portal-license-required)
+By default, the chart would expose MDCB service at `mdcb-svc-tyk-control-plane-tyk-mdcb.tyk.svc:9091`. If you would like to access the control plane from outside the cluster, please change the service type `tyk-mdcb.mdcb.service.type` to NodePort or LoadBalancer.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Should this section be placed after the `Follow the installation output` below.
As service name is based on the namespace and name of your helm release.

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 5]
```diff
@@ -202,26 +197,36 @@ This helm chart enables the `PodDisruptionBudget` for MongoDB with an arbiter re
 Increase the replica count in the helm chart deployment to a minimum of 2 to remedy this issue.
 {{< /note >}}
 
+Configure `global.mongo.mongoURL` and `global.storageType` as below. You should replace password in the connection string with the MONGODB_ROOT_PASSWORD you obtain from the installation output notes.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
Configure `global.mongo.mongoURL` and `global.storageType` as below. You should replace `pass` in the connection string with the MONGODB_ROOT_PASSWORD you obtain from the installation output notes.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 6]
```diff
@@ -268,9 +273,9 @@ Declaring values for such fields as plain text might not be desired. Instead, fo
 
 This section describes how to use Kubernetes secrets to declare confidential fields.
 
-***Tyk Dashboard Admin***
+***Tyk Dashboard and Developer Portal Admin***
 
-If Tyk Dashboard bootstrapping is enabled, a Tyk Dashboard admin user will be created according to the `global.adminUser` field.
+If Tyk Dashboard or Developer Portal bootstrapping is enabled, the admin user will be created according to the `global.adminUser` field.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Admin user is created only when Dashboard bootstrapping is enable.

**Review by komalsukhani — DISMISSED** [review]

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md (hunk 4 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Good point. I've made this change: 
"By default, the chart would expose MDCB service as ClusterIP. If you would like to access the control plane from outside the cluster, please change the service type `tyk-mdcb.mdcb.service.type` to NodePort or LoadBalancer."

I want user to update the relevant setting before they run helm install.

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md (hunk 6 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: @komalsukhani what happens if I have already bootstrapped Dashboard, but run helm upgrade to bootstrap dev portal?

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md [hunk 7]
```diff
@@ -34,21 +34,74 @@ For a quick start guide, please see [deploy hybrid gateway]({{<ref "tyk-cloud/en
...
 
 {{< img src="/img/tyk-charts/tyk-cloud-deployment.png" alt="tyk-cloud-deployment" >}}
 
+## Obtain Remote Control Plane Connection Details from tyk-control-plane Chart
+
+For Tyk Self-Managed MDCB users who want to deploy data planes, you can obtain MDCB connection details from the notes
+of tyk-control-plane installation output.
+
+```
+=== Tyk Control Plane Details ===
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md:54]: Why formatted is this section formatted in between ```?

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md [hunk 8]
```diff
@@ -34,21 +34,74 @@ For a quick start guide, please see [deploy hybrid gateway]({{<ref "tyk-cloud/en
...
+    
+    kubectl create namespace tyk-dp
+
+    kubectl create secret generic tyk-data-plane-details \
+    --from-literal "orgId=$ORG_ID" \
+    --from-literal "userApiKey=$USER_API_KEY" \
+    --from-literal "groupID=$GROUP_ID" \
+    --namespace tyk-dp
+
+3- Refer this Kubernetes secret (tyk-data-plane-details) while installing worker gateways through
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md]: If this is in installation output then text needs updating there also 

```suggestion
3- Refer to this Kubernetes secret (tyk-data-plane-details) while installing worker gateways through
```

**Review by dcs3spp — COMMENTED** [review]
Hi @caroltyk I have reviewed and have left a question relating to potential installation output

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md (hunk 7 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md:54]: It is the terminal output. Should I remove it?

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md [hunk 9]
```diff
@@ -58,24 +111,28 @@ To install the chart from the Helm repository in namespace `tyk` with the releas
...
 
+At a minimum, modify values.yaml for the following settings:
+1. [Set Redis connection details](#set-redis-connection-details-required)
+
+
 Consult the [Configuration](#configuration) section for the available configuration options and modify your local `values.yaml` file accordingly. Then install the chart by issuing the following command below:
 
 ```bash
-    helm install tyk-data-plane tyk-helm/tyk-data-plane -n tyk --create-namespace -f values.yaml
+    helm install tyk-data-plane tyk-helm/tyk-data-plane -n tyk-dp --create-namespace -f values.yaml
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md]:
```suggestion
helm install tyk-data-plane tyk-helm/tyk-data-plane -n tyk-dp --create-namespace -f values.yaml
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md [hunk 10]
```diff
@@ -34,21 +34,74 @@ For a quick start guide, please see [deploy hybrid gateway]({{<ref "tyk-cloud/en
...
 
 1. Go to Deployment tab and create a Hybrid data plane configuration. You can also select from an existing one.
 2. Copy Key, Org ID, and Data Planes Connection String (MDCB) as `global.remoteControlPlane`'s `userApiKey`, `orgId`, and `connectionString` respectively.
 
 {{< img src="/img/tyk-charts/tyk-cloud-deployment.png" alt="tyk-cloud-deployment" >}}
 
+## Obtain Remote Control Plane Connection Details from tyk-control-plane Chart
+
+For Tyk Self-Managed MDCB users who want to deploy data planes, you can obtain MDCB connection details from the notes
+of tyk-control-plane installation output.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md]:
```suggestion
of tyk-control-plane installation output, as listed below.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md (hunk 7 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md:54]: Thanks Carol, I've updated sentence above and have resolved the comment

**Review by dcs3spp — APPROVED** [review]
@caroltyk PR LGTM and approved. Once @buraksekili has approved is the PR ready for release? I think there was a comment from @komalsukhani relating to dashboard admin on bootstrap? 

Should this be released to 5.x or only 5.3?

**Review by buraksekili — APPROVED** [review]
lgtm

**Pull request merged**